core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -417,7 +417,7 @@ object StoreUtils {
     // if OVERFLOW has been provided, then use HEAPPERCENT as the default
     // eviction policy (unless overridden explicitly)
     val hasOverflow = parameters.get(OVERFLOW).map(_.toBoolean)
-        .getOrElse(!isRowTable && !parameters.contains(EVICTION_BY))
+        .getOrElse(!parameters.contains(EVICTION_BY))
```

Comment:
I think this can be changed to just getOrElse(true) because we always want overflow as the only eviction action (and the default one if not explicitly specified).